crates/forge/src/cmd/build.rs

```diff
@@ -228,6 +209,50 @@ impl BuildArgs {
             Ok([config.src, config.test, config.script, foundry_toml])
         })
     }
+
+    /// Check soldeer.lock file consistency using soldeer_core APIs
+    async fn check_soldeer_lock_consistency(&self, config: &Config) {
+        let soldeer_lock_path = config.root.join("soldeer.lock");
+        if !soldeer_lock_path.exists() {
+            return;
+        }
+
+        let lockfile = match soldeer_core::lock::read_lockfile(&soldeer_lock_path) {
+            Ok(lock) => lock,
+            Err(_) => return,
+        };
+
+        let deps_dir = config.root.join("dependencies");
+        for entry in &lockfile.entries {
+            let dep_name = entry.name();
+            let dep_path = deps_dir.join(format!("{}-{}", dep_name, entry.version()));
+
+            if !dep_path.exists() {
+                continue;
+            }
+
+            // Use soldeer_core's integrity check
+            match soldeer_core::install::check_dependency_integrity(entry, &deps_dir).await {
+                Ok(status) => {
+                    use soldeer_core::install::DependencyStatus;
+                    // Check if status indicates a problem
+                    if matches!(status, DependencyStatus::Missing) {
+                        sh_warn!("Dependency '{}' integrity check failed: {:?}", dep_name, status)
+                            .ok();
+                        continue;
+                    }
+                }
+                Err(e) => {
+                    sh_warn!("Dependency '{}' integrity check error: {}", dep_name, e).ok();
+                    continue;
+                }
+            }
```

Comment:
You should also handle the `DependencyStatus::FailedIntegrity` variant here, as it indicates the dependency is present but the file structure or commit hash do not match the lockfile reference hash.